src/mame/includes/8080bw.h

```diff
@@ -177,6 +186,11 @@ class _8080bw_state : public mw8080bw_state
 	DECLARE_READ8_MEMBER(invmulti_eeprom_r);
 	DECLARE_WRITE8_MEMBER(invmulti_eeprom_w);
 	DECLARE_WRITE8_MEMBER(invmulti_bank_w);
+	DECLARE_WRITE8_MEMBER(cane_sh_port_1_w);
+	DECLARE_WRITE8_MEMBER(cane_music_w);
+	DECLARE_WRITE8_MEMBER(cane_unknown_port0_w);
+	DECLARE_WRITE8_MEMBER(cane_76477_en_w);
+	DECLARE_WRITE8_MEMBER(cane_76477_dis_w);
```

Comment:
Please don't add more stuff like this to the state class.  It's already too big, convoluted and poorly encapsulated.  Create a device to encapsulate the audio system for these games and instantiate it.  See the work I've done on mw8080bw.cpp this month for examples.

Reply:
I tried to follow the direction traced by the coders that added other games to that driver in the past 
(Eg:
	DECLARE_READ8_MEMBER(shuttlei_ff_r);
	DECLARE_WRITE8_MEMBER(shuttlei_ff_w);
	DECLARE_WRITE8_MEMBER(shuttlei_sh_port_1_w);
	DECLARE_WRITE8_MEMBER(shuttlei_sh_port_2_w);
)
but I could understant that code rules may change while the project evolve so I'm gonna recode this sound device.

Reply:
Yes, like many MAME drivers, this code was ported from C to C++ with minimal effort by just moving all the functions and static variables into one big class.  It hasn’t been cleaned up since, and more stuff has been hacked into it, making it even harder to understand and more effort to clean up.

The sound system for these new sets is fairly self-contained.  It should be easy to implement it as a device.

Reply:
I've never looked at MAME source code before and I've noticed that some other drivers have a more object oriented approach than others. When I've started coding that driver I was in doubt and lastly I choosed to implement the needed changes keeping the approach already present in the class.
Not a problem to implement it as a device. I'm going to do it as soon as I've some spare time this week. 

Thank you for your support.

Reply:
I refactored the "Cane" related code creating an audio device to encapsulate the audio system as requested.
I also refactored "Cane" and "Orbite" creating their own classes like you've done in mw8080bw

In order to move "Orbite" in its own class I had to move from private to protected some attributes of _8080bw_state (Eg: m_palette) and methods (Eg: clear_extra_columns and set_8_pixels)
I also had to add a new header file in audio/8080bw.h. Surprisingly the "8080-based black and white sound hardware" doesn't had a header file. For this file I didn't know how to format the head discalimer, so I might have done something in the wrong way, let me know if corrections are needed.

Lastly I've got a curiosity: I don't understand the need to have to different drivers for 8080bw and mw8080bw hardware. Why they can't be merged? They share the same hardware and architecture.

Reply:
The reasoning is that mw8080bw.cpp is for original Midway games, and 8080bw.cpp is for other games on similar hardware.  Ideally we'd be able to turn more common things into devices and not have as many dependencies between state classes.  When files or classes get too big, they become harder to understand.  A lot of MAME is only the way it is because we didn't have the modern framework in place when it was written, and it hasn't been updated yet.